Orange/widgets/data/owselectrows.py

```diff
@@ -269,12 +272,31 @@ def _get_value_contents(box):
                 raise TypeError('Type %s not supported.' % type(child))
         return tuple(cont)
 
-    class QDoubleValidatorEmpty(QDoubleValidator):
+    class NumericValidator(QDoubleValidator):
+        def __init__(self):
+            super().__init__()
+            self._validators = []
+
+            validator = QDoubleValidator()
+            validator.setLocale(_DEFAULT_LOCALE_NUMERIC)
+            self._validators.append(validator)
+
+            validator = QDoubleValidator()
+            locale = QLocale(QLocale.C)
+            locale.setNumberOptions(QLocale.RejectGroupSeparator)
+            validator.setLocale(locale)
+            self._validators.append(validator)
+
         def validate(self, input_, pos):
             if not input_:
                 return (QDoubleValidator.Acceptable, input_, pos)
-            else:
-                return super().validate(input_, pos)
+            for validator in self._validators:
```

Comment:
This looks complicated. I would just keep the validation using _DEFAULT_LOCALE_NUMERIC (the way numbers look in all spin boxes). And you can then simply setLocale in __init__ and skip the validator-inside-validator dance.